lib/test/service.go

```diff
@@ -134,3 +144,171 @@ func GetServiceFromKNServiceDescribe(r *KnRunResultCollector, serviceName string
+
+	for _, fn := range options {
+		fn(list)
+	}
+
+	return list
+}
+
+//WithServices appends the given service to ServiceList
+func WithServices(svc *servingv1.Service) ExpectedServiceListOption {
+	return func(list *servingv1.ServiceList) {
+		list.Items = append(list.Items, *svc)
+	}
+}
+
+//GetRevisionListWithOptions returns RevisionList with options provided
+func GetRevisionListWithOptions(options ...ExpectedRevisionListOption) *servingv1.RevisionList {
+	list := &servingv1.RevisionList{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "List",
+		},
+	}
+
+	for _, fn := range options {
+		fn(list)
+	}
+
+	return list
+}
+
+//GetKNExportWithOptions returns Export object with the options provided
+func GetKNExportWithOptions(options ...ExpectedKNExportOption) *clientv1alpha1.Export {
+	knExport := &clientv1alpha1.Export{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "client.knative.dev/v1alpha1",
+			Kind:       "Export",
+		},
+	}
+
+	for _, fn := range options {
+		fn(knExport)
+	}
+
+	return knExport
+}
+
+//Cfg builds servingv1.Configuration with the options provided
+func Cfg(co ...servingtest.ConfigOption) *servingv1.ConfigurationSpec {
+	c := &servingv1.Configuration{
+		Spec: servingv1.ConfigurationSpec{
+			Template: servingv1.RevisionTemplateSpec{
+				Spec: *revisionSpec.DeepCopy(),
+			},
+		},
+	}
+	for _, opt := range co {
+		opt(c)
+	}
+	c.SetDefaults(context.Background())
+	return &c.Spec
+}
+
+var revisionSpec = servingv1.RevisionSpec{
+	PodSpec: corev1.PodSpec{
+		Containers: []corev1.Container{{
+			Image: pkgtest.ImagePath("helloworld"),
+		}},
+		EnableServiceLinks: ptr.Bool(false),
+	},
+	TimeoutSeconds: ptr.Int64(300),
+}
+
+//GetSvcWithOptions returns ksvc with options provided
+func GetSvcWithOptions(name string, so ...servingtest.ServiceOption) *servingv1.Service {
+	svc := servingtest.ServiceWithoutNamespace(name, so...)
+	svc.TypeMeta = metav1.TypeMeta{
+		Kind:       "Service",
+		APIVersion: "serving.knative.dev/v1",
+	}
+	svc.Spec.Template.Spec.Containers[0].Resources = corev1.ResourceRequirements{}
+	return svc
+}
+
+//WithTrafficSplit adds route to ksvc
+func WithTrafficSplit(revisions []string, percentages []int, tags []string) servingtest.ServiceOption {
+	return func(svc *servingv1.Service) {
+		var trafficTargets []servingv1.TrafficTarget
+		for i, rev := range revisions {
+			trafficTargets = append(trafficTargets, servingv1.TrafficTarget{
+				Percent: ptr.Int64(int64(percentages[i])),
+			})
+			if tags[i] != "" {
+				trafficTargets[i].Tag = tags[i]
+			}
+			if rev == "latest" {
+				trafficTargets[i].LatestRevision = ptr.Bool(true)
+			} else {
+				trafficTargets[i].RevisionName = rev
+				trafficTargets[i].LatestRevision = ptr.Bool(false)
+			}
+		}
+		svc.Spec.RouteSpec = servingv1.RouteSpec{
+			Traffic: trafficTargets,
+		}
+	}
+}
+
+//GetRev returns Revision object with the options provided
+func GetRev(name string, options ...servingtest.RevisionOption) servingv1.Revision {
```

Comment:
above, `Cfg` is used to build the configuration and here `GetRev`; all these builder utils should follow the same pattern, IMO we should avoid short names as well. 
wdyt about `BuildRevision` / `BuildConfiguration`?
I'd also avoid `Get` verb here as that denote to client GET call.